plugin/advisor/mysql/advisor_naming_primary_key_convention.go

```diff
@@ -0,0 +1,168 @@
+}
+
+// NamingPKConventionAdvisor is the advisor checking for primary key naming convention.
+type NamingPKConventionAdvisor struct {
+}
+
+// Check checks for primary key naming convention.
+func (check *NamingPKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, err := api.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingPKConventionChecker{
+		level:        level,
+		format:       format,
+		templateList: templateList,
+	}
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type namingPKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	format       string
+	templateList []string
+}
+
+func (checker *namingPKConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.Internal,
+				Title:   "Internal error for primary key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.index) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.NamingPKConventionMismatch,
+				Title:   "Mismatch primary key naming convention",
+				Content: fmt.Sprintf("%q mismatches primary key naming convention, expect %q but found %q", in.Text(), checker.format, indexData.index),
+			})
+		}
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return in, false
+}
+
+func (checker *namingPKConventionChecker) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
+}
+
+// getMetaDataList returns the list of primary key with meta data.
+func (checker *namingPKConventionChecker) getMetaDataList(in ast.Node) []*indexMetaData {
+	var res []*indexMetaData
+
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
+		for _, constraint := range node.Constraints {
+			switch constraint.Tp {
+			case ast.ConstraintPrimaryKey:
```

Comment:
ditto
use `if` instead of `switch case`

Reply:
Maybe we should skip PK naming check for the `CREATE TABLE` statement.

Reply:
Seems we still need to check the naming if the statement is something like `CREATE TABLE tech_book(id INT, name VARCHAR(20), PRIMARY KEY pk_tech_book_name (name))`